pkg/reducer/examples/sum/main.go

```diff
@@ -9,8 +9,17 @@ import (
 	"github.com/numaproj/numaflow-go/pkg/reducer"
 )
 
+// SumReducer implements the reducer.ReducerCreator interface which creates a reducer
+type SumReducer struct {
```

Comment:
The struct name is confusing - should not use `xxxReducer`, that sounds like it's an instance of `Reducer` interface.

Reply:
will update it to `SumReducerCreator`